### DISCUSSION TITLE
Invitation to translate Sumatra?

### DISCUSSION THREAD

**Discussion by hoghoj1** [sumatrapdfreader/sumatrapdf#4112] [Ideas]
Make a invitation to translate somewhere under Options/Change Language. I would love to make it possible to have a **Faroese translation**. This even the population is only 55 thousand!

**Comment by ikanakova** [comment]
Just because Sumatra will be translated into your language does not mean you will have it in that language and not English, see #3867 But it's true that Czechs, Slovaks are few, so maybe for 55 thousand it's worth fixing this bug.

**Comment by hoghoj1** [comment]
Please don't misunderstand, I am only interested in having the possibility to translate it **myself for FREE.** I have translated 2 other applications before, a faroese option in the webbuilder **Typesettercms** and a faroese edition of the danish program **CD-ORD (for Dysletics)**. I like Sumatra and would like to offer my fellow citizens to enjoy this program in their own language. :-)

**Reply by ikanakova** [reply]
> Please don't misunderstand, I am only interested in having the possibility to translate it **myself for FREE.** I have translated 2 other applications before, a faroese option in the webbuilder **Typesettercms** and a faroese edition of the danish program **CD-ORD (for Dysletics)**. I like Sumatra and would like to offer my fellow citizens to enjoy this program in their own language. :-)

Here is the translation page: https://www.apptranslator.org/app/SumatraPDF

But I don't know how to ask for new language. But I would try to open a new issue and write that your language is missing on the mentioned page.

**Comment by kjk** [comment]
@hoghoj1  Thanks for the interest in translating to Faroese.

I've added Faroese language: https://www.apptranslator.org/app/SumatraPDF/fo

I periodically download the translations from the server and update the app.

Once a number of translations crosses a certain threshold, it shows as an option (i.e. we don't want to show half-translated languages as an option).

Please keep in mind that I periodically add new strings so keeping translations up-to-date is a process.

If you use RSS reader I suggest subscribing to https://www.apptranslator.org/rss?app=SumatraPDF&lang=fo

This will show when there are new strings to translate.

I also pre-announce a new release at lest 2 weeks in advance, to give people a chance to update translations before a release.

**Comment by hoghoj1** [comment]
Hi Krzysztof,
I'm not a programmer and have an issue with how to use the translation 
page. I click on "Add a translation" and nothing happens else that an 
"#" is being added to the link ???

------ Original Meddelelse ------
Fra "Krzysztof Kowalczyk" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" ***@***.***>
Cc "hoghoj1" ***@***.***>; "Mention" ***@***.***>
Dato 29.02.2024 14:06:48
Emne Re: [sumatrapdfreader/sumatrapdf] Invitation to translate Sumatra? 
(Discussion #4112)

**Comment by kjk** [comment]
Did you log-in with your GitHub account using the link in the upper right? That's the first step.

Then when you click on a 'Add a translation` link, this should happen:

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/7935/30de6b06-3195-4741-85bd-f615e8c95790)

If not, which browser are you using?

I just checked that it works in Chrome in Windows.

**Comment by hoghoj1** [comment]
Hi Krzysztof,
yes I did log in in the upper right but nothing happened when I clicked 
the translation link and yes, I use Chrome too.
Then I waited for your answer and tried again and now it works - I'm 
starting translation :-D, thx again!

------ Original Meddelelse ------
Fra "Krzysztof Kowalczyk" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" ***@***.***>
Cc "hoghoj1" ***@***.***>; "Mention" ***@***.***>
Dato 29.02.2024 14:50:54
Emne Re: [sumatrapdfreader/sumatrapdf] Invitation to translate Sumatra? 
(Discussion #4112)

**Comment by hoghoj1** [comment]
Hi Krzysztof,
When I login I see an error which I can't recognize, I'll send you a 
screenshot. Now nothing happens when I click on "Add Translation". I've 
tried Chrome, Edge and Firefox-same result"  ???

------ Original Meddelelse ------
Fra "Høgni Højgaard" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" 
***@***.***>
Dato 29.02.2024 16:57:10
Emne Re[2]: [sumatrapdfreader/sumatrapdf] Invitation to translate 
Sumatra? (Discussion #4112)

**Comment by hoghoj1** [comment]
OOps - forgot the screenshot - here it is  :-)

------ Original Meddelelse ------
Fra "Høgni Højgaard" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" 
***@***.***>
Dato 03.03.2024 12:44:31
Emne Re[3]: [sumatrapdfreader/sumatrapdf] Invitation to translate 
Sumatra? (Discussion #4112)

**Reply by kjk** [reply]
still no screenshot

**Comment by GitHubRulesOK** [comment]
@hoghoj1 
I am impressed that you have done so many
But the way the `&` works needs to be directly before the letter that will become the shortcut so I noted there is a space here that could be removed before the `B`  (also a space before `H` at `&Continue installing 32-bit version => & Halt áfram at installera 32-bita útgávu`
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/68842624-72a0-477f-93ba-4996a8780c80)

**Comment by hoghoj1** [comment]
OK, but the error reported also makes it impossible to scroll down and 
edit mistakes. If that is causing the error, is it then possible for you 
to remove that space?

------ Original Meddelelse ------
Fra "GitHubRulesOK" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" ***@***.***>
Cc "hoghoj1" ***@***.***>; "Mention" ***@***.***>
Dato 03.03.2024 14:30:19
Emne Re: [sumatrapdfreader/sumatrapdf] Invitation to translate Sumatra? 
(Discussion #4112)

**Reply by GitHubRulesOK** [reply]
@hoghoj1 
I dont have an error so should be able to adjust those 2 for you, and will do so. **[edit] Now just done the 2**
However it would be best going forwards if you show or type the error you are getting.

**Comment by GitHubRulesOK** [comment]
@kjk 
I hit an error with cut and paste "string cannot be empty" which needed fiddle the last character, so I wonder if there is some extra information could be added like "Re-check entry before submit"

Minor pedantic issue  with list of names (;-) my git hub handle is currently `https://www.apptranslator.org/user/GitHubRulesOK` but I previously had other submissions can those be reassigned from
 `https://www.apptranslator.org/user/throwaway10261` made 6 translations
`https://www.apptranslator.org/user/throwaway1026` made 13 translations

**Comment by hoghoj1** [comment]
Hi Krzysztof,
actually I have only managed to see an error once and only in the 
screenshot I sent you. I'll send 2 more here, go in on the address 
https://www.apptranslator.org/app/SumatraPDF/fo, log in on GitHub 
successfully, choose Faroese successfully, try to click on several Add 
Translation and there is no response at all! :-(

------ Original Meddelelse ------
Fra "GitHubRulesOK" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" ***@***.***>
Cc "hoghoj1" ***@***.***>; "Mention" ***@***.***>
Dato 03.03.2024 14:51:32
Emne Re: [sumatrapdfreader/sumatrapdf] Invitation to translate Sumatra? 
(Discussion #4112)

**Comment by hoghoj1** [comment]
Hi Krzysztof,
now it works again - no problem :-) I'll contact you if further issues 
appear, ok?

------ Original Meddelelse ------
Fra "Høgni Højgaard" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" 
***@***.***>
Dato 03.03.2024 15:24:28
Emne Re[2]: [sumatrapdfreader/sumatrapdf] Invitation to translate 
Sumatra? (Discussion #4112)

**Comment by hoghoj1** [comment]
Hi Krzysztof,
sorry to bother you again, but here is an error after I cleansed the PC 
for everything, then tried to log in again ?

------ Original Meddelelse ------
Fra "Høgni Højgaard" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" 
***@***.***>
Dato 03.03.2024 15:43:38
Emne Re[3]: [sumatrapdfreader/sumatrapdf] Invitation to translate 
Sumatra? (Discussion #4112)

**Comment by hoghoj1** [comment]
Hi Krzysztof,
beginning to see a pattern, starting all over I'm allowed to translate 4 
items then it stops!

------ Original Meddelelse ------
Fra "Høgni Højgaard" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" 
***@***.***>
Dato 03.03.2024 16:23:03
Emne Re[4]: [sumatrapdfreader/sumatrapdf] Invitation to translate 
Sumatra? (Discussion #4112)

**Reply by kjk** [reply]
please post a screenshot of a  what happens. you must do it via web interface, not by email

I still have not idea what problem you're seeing.

**Comment by hoghoj1** [comment]
Ok, I will do that. How do I put it on Web, I have not tried that before?

søn. 3. mar. 2024 19.06 skrev Krzysztof Kowalczyk ***@***.***

**Reply by GitHubRulesOK** [reply]
@hoghoj1 
in a web browser
https://github.com/sumatrapdfreader/sumatrapdf/discussions/4112#discussioncomment-8659343 should bring you to the comment above where you can  "Write a reply" and copy past text or images etc.

**Reply by kjk** [reply]
Use web-interface i.e. https://github.com/sumatrapdfreader/sumatrapdf/discussions/4112

You can:
* drag&drop screenshot file onto a reply text field
* if screenshot is in the clipboard, you can `Ctrl + V` in reply text field  (on windows) to add image

**Reply by kjk** [reply]
or you can e-mail it to me at `kkowalczyk@gmail.com`

**Reply by hoghoj1** [reply]
![error-App translator](https://github.com/sumatrapdfreader/sumatrapdf/assets/44140864/9ae93dc1-b20b-48cc-becf-022277f7bf32)
![error-App translator_2](https://github.com/sumatrapdfreader/sumatrapdf/assets/44140864/41856761-3e08-43b9-98da-b8f3fe76876b)
![error-App translator_3](https://github.com/sumatrapdfreader/sumatrapdf/assets/44140864/48f6e59a-5c11-458e-a635-ca10ae80cee9)
![error-App translator_4](https://github.com/sumatrapdfreader/sumatrapdf/assets/44140864/b6016a62-e318-4fd6-abfd-29f8af404d88)

**Reply by hoghoj1** [reply]
I managed to do this :-)

**Reply by kjk** [reply]
I've never seen such error, no one else ever reported it and I can't reproduce it right now.

I also don't see anything in the server logs.

My only guess is there's something weird about your network connection, maybe using a vpn, non-standard browser configuration (maybe disabled cookies?).

This error comes from login sequence but if you're logged in that shouldn't even be invoked.

So:
* try logout and re-login
* try different network connection
* try a different browser
* if you have some non-standard aggressive settings or browser extensions, try disabling those extensions / reverting settings

**Comment by hoghoj1** [comment]
Hi Krzysztof,
yes I also have never experienced such trouble before. I'm not using a 
VPN and cookies are enabled. I cleaned the Chrome browser, restarted the 
PC after disabling some unneccesary upstart services and started a new 
session with Chrome - it didn't work. Started an Incognito session 
(Chrome) and managed 2 transl. - it stopped again. Logged out and 
installed a brand new Firefox browser - seems to work now having made 10 
transl. and 1 edit session. Going to sleep now and will start again 
tomorrow morning hoping it will be ok. I intend to finish the work this 
coming week :-)

------ Original Meddelelse ------
Fra "Krzysztof Kowalczyk" ***@***.***>
Til "sumatrapdfreader/sumatrapdf" ***@***.***>
Cc "hoghoj1" ***@***.***>; "Mention" ***@***.***>
Dato 03.03.2024 22:01:03
Emne Re: [sumatrapdfreader/sumatrapdf] Invitation to translate Sumatra? 
(Discussion #4112)

**Comment by hoghoj1** [comment]
Hi Krzysztof, I've finished translating the Faroese edition 98%. 4 items are untranslated because we use same words. It's now up to you when you update the system, but please let me know :-)

**Reply by kjk** [reply]
Thanks. Next [pre-release build](https://www.sumatrapdfreader.org/prerelease) should have the translations and should be available in about 24hr.

Try pre-release if version is greater than `15976`

**Reply by GitHubRulesOK** [reply]
I bumped those 4 as "Just the Same" to make the position cleaner for any future round.

**Reply by hoghoj1** [reply]
Hi Krzysztof, I'm testing my translation now in build 15979 and have found a few grammatic errors in the 'Go To' menu.
![næsta-næstu](https://github.com/sumatrapdfreader/sumatrapdf/assets/44140864/d1081873-047b-4777-8335-b9de7f37267e)
Faroese language uses different words for 'next' depending on whether what you intend to say. If you say: 'Here is the next page' in Faroese that is 'Her er næst**a** síð**a**'. If you say 'Go to next page' in Faroese that is 'Far til næst**u** síð**u**.
In the translator app I find 2 items 'Next Page' and '&Next Page', will both items have to be re-edited?

**Reply by kjk** [reply]
Yes, from the point of view of translation system, those are completely different strings.

**Reply by hoghoj1** [reply]
Can I re-edit those strings?

**Reply by kjk** [reply]
Sure, click `Edit` link in apptranslator

**Comment by hoghoj1** [comment]
I've downloaded the preview release build 15980, and all the & links **_are absent!_** Why?

**Reply by GitHubRulesOK** [reply]
& need the Alt key first to trigger the underline display. So Alt+F triggers the File menu

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/06c76f98-b718-4677-8fef-cf407dfe4c8f)  

 You should also try to remove conflicts of 2 letters have same underline (not easy) so for example in English
Move Minimize and Maximize all have different KEY letters

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/e1d4af92-3612-4763-aa2b-73a8d9b54ee7)

**Reply by hoghoj1** [reply]
Of course, I was just foolish enough to forget that. I just have reedited all those links, so they are different, it's all OK now :-)

**Reply by hoghoj1** [reply]
Have reedited a few &items more - looking forward for the next prerelease :-)

**Comment by sai-vineeth-pasunuru** [comment]
I realized my native language "Telugu" (spoken by 83M native speakers) is not one of the 72 languages supported in the application. I would love to contribute translations so please guide me.

**Reply by GitHubRulesOK** [reply]
@sai-vineeth-pasunuru

With so many languages comes increased overheads so does need a reasonable commitment to be kept up-to-date.

e.g. There are not many Klingon Speakers and whist it might be "fun" to use some odd lingos the aim should be for a local mass usage and even "English Cornish" (Only **563** [L2 users](https://en.wikipedia.org/wiki/Second_language) as of the 2021 Census) meets that criteria ![image](https://github.com/user-attachments/assets/418f2077-58cf-45ba-8d5b-fec31670adcb)
Currently that is updated in bursts and presently falling behind at 82% https://www.apptranslator.org/app/SumatraPDF/kw

A fair number of not well maintained include
```
Languages (sort by: [name](https://www.apptranslator.org/app/SumatraPDF?sort=name), untranslated)

[Burmese](https://www.apptranslator.org/app/SumatraPDF/mm) (337 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=mm))
[Shona](https://www.apptranslator.org/app/SumatraPDF/sn) (319 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=sn))
[Afrikaans](https://www.apptranslator.org/app/SumatraPDF/af) (313 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=af))
[Punjabi](https://www.apptranslator.org/app/SumatraPDF/pa) (296 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=pa))
[Javanese](https://www.apptranslator.org/app/SumatraPDF/jv) (282 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=jv))
[Frisian](https://www.apptranslator.org/app/SumatraPDF/fy-nl) (268 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=fy-nl))
[Malayalam](https://www.apptranslator.org/app/SumatraPDF/ml) (237 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=ml))
[Bengali](https://www.apptranslator.org/app/SumatraPDF/bn) (197 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=bn))
[Uzbek](https://www.apptranslator.org/app/SumatraPDF/uz) (149 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=uz))
[Bosnian](https://www.apptranslator.org/app/SumatraPDF/bs) (148 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=bs))
[Albanian](https://www.apptranslator.org/app/SumatraPDF/sq) (145 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=sq))
[Serbian Latin](https://www.apptranslator.org/app/SumatraPDF/sp-rs) (145 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=sp-rs))
[Sinhala](https://www.apptranslator.org/app/SumatraPDF/si) (145 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=si))
[Kurdish](https://www.apptranslator.org/app/SumatraPDF/ku) (142 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=ku))
[Nepali](https://www.apptranslator.org/app/SumatraPDF/ne) (132 untranslated, [rss](https://www.apptranslator.org/rss?app=SumatraPDF&lang=ne))
```

**Reply by sai-vineeth-pasunuru** [reply]
@GitHubRulesOK I understand. Thank you for the response!
